connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/SparkSessionE2ESuite.scala

```diff
@@ -96,5 +103,148 @@ class SparkSessionE2ESuite extends RemoteSparkSession {
+      assert(spark.getTags() == Set("one", "two"))
+      try {
+        spark
+          .range(10)
+          .map(n => {
+            Thread.sleep(30000); n
+          })
+          .collect()
+      } finally {
+        spark.clearTags() // clear for the case of thread reuse by another Future
+      }
+    }(executionContext)
+    val q3 = Future {
+      assert(spark.getTags() == Set())
+      spark.addTag("foo")
+      spark.removeTag("foo")
+      assert(spark.getTags() == Set()) // check that remove works removing the last tag
+      spark.addTag("two")
+      assert(spark.getTags() == Set("two"))
+      try {
+        spark
+          .range(10)
+          .map(n => {
+            Thread.sleep(30000); n
+          })
+          .collect()
+      } finally {
+        spark.clearTags() // clear for the case of thread reuse by another Future
+      }
+    }(executionContext)
+    val q4 = Future {
+      assert(spark.getTags() == Set())
+      spark.addTag("one")
+      spark.addTag("two")
+      spark.addTag("two")
+      assert(spark.getTags() == Set("one", "two"))
+      spark.removeTag("two") // check that remove works, despite duplicate add
+      assert(spark.getTags() == Set("one"))
+      try {
+        spark
+          .range(10)
+          .map(n => {
+            Thread.sleep(30000); n
+          })
+          .collect()
+      } finally {
+        spark.clearTags() // clear for the case of thread reuse by another Future
+      }
+    }(executionContext)
+    val interrupted = mutable.ListBuffer[String]()
+
+    // q2 and q3 should be cancelled
+    interrupted.clear()
+    eventually(timeout(20.seconds), interval(1.seconds)) {
+      val ids = spark.interruptTag("two")
+      interrupted ++= ids
+      assert(interrupted.length == 2, s"Interrupted operations: $interrupted.")
+    }
+    val e2 = intercept[SparkException] {
+      ThreadUtils.awaitResult(q2, 1.minute)
+    }
+    assert(e2.getCause.getMessage contains "OPERATION_CANCELED")
+    val e3 = intercept[SparkException] {
+      ThreadUtils.awaitResult(q3, 1.minute)
+    }
+    assert(e3.getCause.getMessage contains "OPERATION_CANCELED")
+    assert(interrupted.length == 2, s"Interrupted operations: $interrupted.")
+
+    // q1 and q4 should be cancelled
+    interrupted.clear()
+    eventually(timeout(20.seconds), interval(1.seconds)) {
+      val ids = spark.interruptTag("one")
+      interrupted ++= ids
+      assert(interrupted.length == 2, s"Interrupted operations: $interrupted.")
+    }
+    val e1 = intercept[SparkException] {
+      ThreadUtils.awaitResult(q1, 1.minute)
+    }
+    assert(e1.getCause.getMessage contains "OPERATION_CANCELED")
+    val e4 = intercept[SparkException] {
+      ThreadUtils.awaitResult(q4, 1.minute)
+    }
+    assert(e4.getCause.getMessage contains "OPERATION_CANCELED")
+    assert(interrupted.length == 2, s"Interrupted operations: $interrupted.")
+  }
+
+  test("interrupt operation") {
```

Comment:
@juliuszsompolski after this pr, maven test `SparkSessionE2ESuite` start failed:

```
build/mvn clean install -DskipTests -Phive  
build/mvn clean test -pl connector/connect/client/jvm
``` 

```
SparkSessionE2ESuite:
- interrupt all - background queries, foreground interrupt *** FAILED ***
  The code passed to eventually never returned normally. Attempted 30 times over 20.274941708000004 seconds. Last failure message: Some("unexpected failure in q2: org.apache.spark.SparkException: org/apache/spark/sql/connect/client/SparkResult") was not empty Error not empty: Some(unexpected failure in q2: org.apache.spark.SparkException: org/apache/spark/sql/connect/client/SparkResult). (SparkSessionE2ESuite.scala:69)
- interrupt all - foreground queries, background interrupt *** FAILED ***
  "org/apache/spark/sql/connect/client/SparkResult" did not contain "OPERATION_CANCELED" Unexpected exception: org.apache.spark.SparkException: org/apache/spark/sql/connect/client/SparkResult (SparkSessionE2ESuite.scala:99)
- interrupt tag *** FAILED ***
  The code passed to eventually never returned normally. Attempted 30 times over 20.256318458000003 seconds. Last failure message: ListBuffer() had length 0 instead of expected length 2 Interrupted operations: ListBuffer().. (SparkSessionE2ESuite.scala:197)
- interrupt operation *** FAILED ***
  org.apache.spark.SparkException: org/apache/spark/sql/connect/client/SparkResult
  at org.apache.spark.sql.connect.client.GrpcExceptionConverter$.toSparkThrowable(GrpcExceptionConverter.scala:53)
  at org.apache.spark.sql.connect.client.GrpcExceptionConverter$.convert(GrpcExceptionConverter.scala:30)
  at org.apache.spark.sql.connect.client.GrpcExceptionConverter$$anon$1.hasNext(GrpcExceptionConverter.scala:38)
  at org.apache.spark.sql.connect.client.SparkResult.org$apache$spark$sql$connect$client$SparkResult$$processResponses(SparkResult.scala:82)
  at org.apache.spark.sql.connect.client.SparkResult.operationId(SparkResult.scala:173)
  at org.apache.spark.sql.SparkSessionE2ESuite.$anonfun$new$31(SparkSessionE2ESuite.scala:241)
  at org.apache.spark.sql.connect.client.util.RemoteSparkSession.$anonfun$test$1(RemoteSparkSession.scala:235)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  ...

```

Reply:
@LuciferYang 
> Unexpected exception: org.apache.spark.SparkException: org/apache/spark/sql/connect/client/SparkResult

This appears to be a reocurrance of the ClassNotFound that you attempted to fix in https://github.com/apache/spark/pull/41487 by moving these tests to a separate class.
Like I wrote there, it's a bigger problem with UDF serialization pulling in seemingly unrelated classes. I raised https://issues.apache.org/jira/browse/SPARK-43744 on this, and @zhenlineo is working on it (WIP https://github.com/apache/spark/pull/42069)

Reply:
> @LuciferYang
> 
> > Unexpected exception: org.apache.spark.SparkException: org/apache/spark/sql/connect/client/SparkResult
> 
> This appears to be a reocurrance of the ClassNotFound that you attempted to fix in #41487 by moving these tests to a separate class. Like I wrote there, it's a bigger problem with UDF serialization pulling in seemingly unrelated classes. I raised https://issues.apache.org/jira/browse/SPARK-43744 on this, and @zhenlineo is working on it (WIP #42069)

OK

Reply:
@juliuszsompolski  Sorry to bother you, https://github.com/apache/spark/pull/42069 has already been merged, but these 4 test cases will still maven test fail. Do you have any plans or suggestions for fix?

Reply:
Hi @bogao007 it would be great when you get change to fix https://issues.apache.org/jira/browse/SPARK-44576
After the fix, we can move the whole test to use addArtifact to fix the 4 maven tests. I think we need to fix them before the 3.5 release.

Reply:
@zhenlineo So ultimately, we still need to solve this issue by treating `client.jar` or `client-test.jar` as `Artifact`?

Reply:
@LuciferYang Our current solution only stubs classes from user session scoped libs. That covers jars added via `session.addArtifact`. The bug Spark-44576 should be fixed as streaming should not fail when user adds session scoped libs. If the failing tests really bothers you, we can also move them into separate test file for now. The cause is the unnecessary dependency pulled by lambdas inside the same class file.

Reply:
@zhenlineo Thank you for your response. As long as we resolve this issue before the release of 3.5, it will be fine.

Reply:
> @LuciferYang Our current solution only stubs classes from user session scoped libs. That covers jars added via `session.addArtifact`. The bug [Spark-44576](https://issues.apache.org/jira/browse/SPARK-44576) should be fixed as streaming should not fail when user adds session scoped libs. If the failing tests really bothers you, we can also move them into separate test file for now. The cause is the unnecessary dependency pulled by lambdas inside the same class file.

I have created https://issues.apache.org/jira/browse/SPARK-44784 and marked it as `Blocker`. We should fix this issue before the release of Apache Spark 3.5.0. also cc @xuanyuanking to know.